RationaleMCP/0031/differences.md

```diff
@@ -670,98 +670,59 @@ initial equation
-
-Examples:
+The basic Flat Modelica way of expressing a variable's guess value priority take the form of a special kind of initial equation:
 ```
-  parameter equation guess('x', 8) = 1.0; /* Guess value priority for 'x' is 8. */
-  parameter equation guess('arr', 5) = fill(1.0, 3); /* Guess value priority for all elements of the array 'arr' is 5. */
 initial equation
-  guess('y', 7) = 1.5; /* Guess value priority for 'y' is 7. */
-  'z' = guess('y'); /* Use 1 argument form except where priority is specified.  */
+  prioritize('x', 2); /* The guess value priority of 'x' is 2. */
 ```
 
-The above syntax works well for homogeneous priorities, which will be the case for the arrays coming from current full Modelica, but not for records.  For example, here `r.b` has higher priority than `r.a`:
-```
-model M
-  record R
-    Real a(start = 1.1);
-    Real b(start = 1.2);
-  end R;
-  R r(b(start = 1.3));
-end M;
-```
+The second argument of `prioritize` – denoted _priority_ in the grammar – shall be an `Integer` constant.  Lower value means higher priority; that is, when making a choice based on priority, the variable with lower _priority_ value should be given precedence.
 
-The basic way of dealing with the two different priorities would be to use two separate equations:
+Specification of priority is only allowed for components whose guess value parameter is explicitly present in the model.  Example:
 ```
-parameter equation guess('r'.'a', 1) = 1.1;
-parameter equation guess('r'.'b', 0) = 1.3;
+  Real 'x';
+  parameter equation guess('x') = 1.1;
+  Real 'y';
+  Real 'z';
+initial equation
+  guess('y') = 1.2;
+  prioritize('x', 1); /* OK: Mentioned in guess value parameter equation. */
+  prioritize('y', 2); /* OK: Mentioned in initial equation. */
+  prioritize('z', 3); /* Error: Guess value is not explicitly mentioned anywhere. */
 ```
 
-If one would like to avoid splitting into scalar equations, variants that come to mind include:
-```
-parameter equation guess('r', 'R'(1, 0)) = 'R'(1.1, 1.3);
+Array variables are no exception:
 ```
-and
-```
-parameter equation guess('r') = 'R'(guessPriority(1.1, 1), guessPriority(1.3, 0));
+  Real[3] 'x';
+  parameter equation guess('x') = {1.1, 1.2, 1.3};
+initial equation
+  prioritize('x', 2);
 ```
 
-In case all record members have the same priority:
+Multiple specification is an error.  For example (here `R` is a record type):
```

Comment:
In full Modelica records do not have start attibutes. Do we really want to define guess values and prioritx to record variables/values?

Reply:
The corresponding full Modelica model would be something like this:
```
model RecordMemberStart
  record R
    Real x;
  end R;

  R r(x(start = 1.1));
equation
  der(r.x) = 0.0;
end RecordMemberStart;
```
Is that a problem?

Reply:
But this does not explain why we want to this. If we have just one member it seems simpler to give it priority, and if there are they are often given separately.